include/matx/transforms/cub.h

```diff
@@ -1949,6 +2076,127 @@ void sort_impl(OutputTensor &a_out, const InputOperator &a,
 #endif
 }
 
+
+
+/**
+ * Sort rows of a tensor
+ *
+ * Sort rows of a tensor using a radix sort. Currently supported types are
+ * float, double, ints, and long ints (both signed and unsigned). For a 1D
+ * tensor, a linear sort is performed. For 2D and above each row of the inner
+ * dimensions are batched and sorted separately. There is currently a
+ * restriction that the tensor must have contiguous data in both rows and
+ * columns, but this restriction may be removed in the future.
+ *
+ * @note Temporary memory is used during the sorting process, and about 2N will
+ * be allocated, where N is the length of the tensor.
+ *
+ * @tparam T1
+ *   Type of data to sort
+ * @tparam RANK
+ *   Rank of tensor
+ * @param idx_out
+ *   Indices of sorted tensor
+ * @param a
+ *   Input operator
+ * @param dir
+ *   Direction to sort (either SORT_DIR_ASC or SORT_DIR_DESC)
+ * @param exec
+ *   CUDA executor
+ */
+template <typename OutputTensor, typename InputOperator>
+void argsort_impl(OutputTensor &idx_out, const InputOperator &a,
+          const SortDirection_t dir,
+          cudaExecutor exec = 0)
+{
+#ifdef __CUDACC__
+  MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
+
+  static constexpr int RANK = OutputTensor::Rank();
+
+  using a_type = typename InputOperator::value_type;
+  a_type *a_ptr = nullptr;
+  a_type *a_out_ptr = nullptr;
+  index_t *idx_in_ptr = nullptr;
+  detail::tensor_impl_t<a_type, InputOperator::Rank()> tmp_a;
+  detail::tensor_impl_t<a_type, InputOperator::Rank()> tmp_a_out;
+  detail::tensor_impl_t<index_t, InputOperator::Rank()> tmp_idx_in;
+    
+  // sorting currently requires a contiguous tensor view, so allocate a temporary
+  // tensor to copy the input if necessary.
+  bool use_a = false;
+  if constexpr (is_tensor_view_v<InputOperator>) {
+    if (a.IsContiguous()) {
+      make_tensor(tmp_a, a.Data(), a.Shape());
+      use_a = true;
+    }
+  }
+  if (!use_a) {
+    matxAlloc((void**)&a_ptr, TotalSize(a) * sizeof(a_type), MATX_MANAGED_MEMORY, exec.getStream());
```

Comment:
We should try to avoid allocating managed memory internally. This doesn't work on WSL2 and other platforms. Can this be changed to an async device allocation?